### ISSUE TITLE
webp conversion happening before optimizers

### ISSUE THREAD

**Issue by Enalmada** [thumbor/thumbor#1061]
I am totally new to Thumbor and trying out Thumbor 6.4.2 today with lossy optimizers ([mozjpeg](https://github.com/thumbor/thumbor-plugins/wiki/Mozjpeg)).   Just one huge problem, when turning on AUTO_WEBP true it is generating webp images with file sizes bigger than original image.  
original image: 10k
auto_webp false: 4.5k jpg
auto_webp true: 12k webp
expected webp size: 3.5k (based on some online webp conversion of the thumbor optimized jpg)

### Thumbor request URL
http://localhost:8000/unsafe/200x500/smart/http://user-image.logdown.io/user/4124/blog/4163/post/152158/bfMZCfHnSeOo4vN1pClp_christina.jpg

### Expected behaviour
webp conversion should happen after optimizers so that lossy benefits are applied.  final auto true webp image size is smaller than auto false jpg.

### Actual behaviour
I am just guessing webp conversion appears to be happening before optimizers (so they are not being run because file type changes to webp before lossy png optimizer step can work the magic)

### Operating system
Amazon Linux

### Your thumbor.conf (everything else is default I believe)
AUTO_WEBP = True
OPTIMIZERS = [
    'thumbor_plugins.optimizers.pngquant',
    'thumbor_plugins.optimizers.mozjpeg'
]
DETECTORS = [
    'thumbor.detectors.face_detector',
    'thumbor.detectors.profile_detector'
]

**Comment by Enalmada** [comment]
Here are the debug logs.  I can see MOZJPEG not taking effect for AUTO_WEBP

AUTO_WEBP = True
```
2018-05-07 18:44:52 thumbor:DEBUG Image format set by AUTO_WEBP as .webp.
2018-05-07 18:44:52 thumbor:DEBUG Content Type of image/webp detected.
```

AUTO_WEBP = False
```
2018-05-07 18:46:30 thumbor:DEBUG No image format specified. Retrieving from the image extension: .jpg.
2018-05-07 18:46:30 thumbor:DEBUG Content Type of image/jpeg detected.
2018-05-07 18:46:30 thumbor:DEBUG [MOZJPEG] running: /usr/local/bin/cjpeg -quality 75 -optimize /tmp/tmp7I1b_9-intermediate > /tmp/tmp7I1b_9
2018-05-07 18:46:30 tornado.access:INFO 200 GET /unsafe/200x500/smart/user-image.logdown.io/user/4124/blog/4163/post/152158/bfMZCfHnSeOo4vN1pClp_christina.jpg (184.187.165.208) 62.38ms
```

**Comment by kkopachev** [comment]
Ran your image through local thumbor and even at original resolution, resulted webp image is around 6k
http://localhost:8888/unsafe/800xorig/smart/http://user-image.logdown.io/user/4124/blog/4163/post/152158/bfMZCfHnSeOo4vN1pClp_christina.jpg

For 200x500 it gives 2872 bytes as webp.

mozjpeg in general are lossy, so you don't want to run image through multiple steps of recompression and quality loss. Hence raw image bytes go straight to webp encoder.

**Comment by Enalmada** [comment]
No fair!!!  Thank you for doing that.  @*#& I guess there is something wrong with how webp is being generated on my thumbor install  I am on centos and I noticed none of my filters work either despite pip install thumbor running without any apparent issue.   

![image](https://user-images.githubusercontent.com/1892132/39795272-e3c5cfaa-5303-11e8-9f22-c10f2f143280.png)
![image](https://user-images.githubusercontent.com/1892132/39795313-26fdcf16-5304-11e8-997b-14a23f7f6cfe.png)

**Comment by Enalmada** [comment]
So I think my real problem is actually that filters are not installed.  For example, I can't do blur even though blur is defined in my filter list.  Now thanks to kkopachev I suspect webp not working is the same problem.  I am running on amazon linux (because I wanted to eventually put this on a beanstalk instance).   Can anyone suggest why or what I could try to get my filters installed/working before I give up on amazon linux? 

```
[root@ip-172-31-23-70 libwebp-1.0.0]# pip install thumbor
WARNING: Running pip install with root privileges is generally not a good idea. Try `pip install --user` instead.
Collecting thumbor
Requirement already satisfied: libthumbor>=1.3.2 in /usr/lib/python2.7/site-packages (from thumbor)
Requirement already satisfied: pycurl<7.44.0,>=7.19.0 in /usr/lib64/python2.7/site-packages (from thumbor)
Requirement already satisfied: pycryptodome>=3.4.7 in /usr/lib64/python2.7/site-packages (from thumbor)
Requirement already satisfied: argparse in /usr/lib/python2.7/site-packages (from thumbor)
Requirement already satisfied: tornado<5.0.0,>=4.1.0 in /usr/lib64/python2.7/site-packages (from thumbor)
Requirement already satisfied: webcolors in /usr/lib/python2.7/site-packages (from thumbor)
Requirement already satisfied: futures in /usr/lib/python2.7/site-packages (from thumbor)
Requirement already satisfied: Pillow<5.0.0,>=3.0.0 in /usr/lib64/python2.7/site-packages (from thumbor)
Requirement already satisfied: piexif<1.1.0,>=1.0.13 in /usr/lib/python2.7/site-packages (from thumbor)
Requirement already satisfied: six in /usr/lib/python2.7/site-packages (from thumbor)
Requirement already satisfied: statsd>=3.0.1 in /usr/lib/python2.7/site-packages (from thumbor)
Requirement already satisfied: pytz in /usr/lib/python2.7/site-packages (from thumbor)
Requirement already satisfied: derpconf>=0.2.0 in /usr/lib/python2.7/site-packages (from thumbor)
Requirement already satisfied: certifi in /usr/lib/python2.7/site-packages (from tornado<5.0.0,>=4.1.0->thumbor)
Requirement already satisfied: singledispatch in /usr/lib/python2.7/site-packages (from tornado<5.0.0,>=4.1.0->thumbor)
Requirement already satisfied: backports-abc>=0.4 in /usr/lib/python2.7/site-packages (from tornado<5.0.0,>=4.1.0->thumbor)
Requirement already satisfied: olefile in /usr/lib/python2.7/site-packages (from Pillow<5.0.0,>=3.0.0->thumbor)
Installing collected packages: thumbor
Successfully installed thumbor-6.4.2
```

**Comment by Enalmada** [comment]
I reinstalled on ubuntu following these directions:
https://gist.github.com/Enalmada/3e68169e1689d03e4412af23439932c6

Filters work now!  Webp was still 12k.  Then I uncommented QUALITY=85 and WEBP_QUALITY=80 and it suddenly started working as expected.   So moral of this story is use ubuntu and make sure you explicitly set values you want as commented out defaults are not necessarily set to what you might think.
